### ISSUE TITLE
`editoryInlayHint.foreground/background` doesn't apply when extension is enabled

### ISSUE THREAD

**Issue by crd2333** [catppuccin/vscode#400]
I'm not sure if I have done something wrong, but it seems the color for **inlay hint** didn't take effect, in contrast to **built-in themes**.

Settings for theme Macchiato(one of the setting is useful, though):
![1](https://github.com/catppuccin/vscode/assets/113617021/306f678f-1144-475c-9e6f-a9404b5e9049)
Settings for built-in theme:
![2](https://github.com/catppuccin/vscode/assets/113617021/8861de58-7030-457f-a762-198e21510b06)

**Comment by sgoudham** [comment]
Hey :wave:,

Thanks for raising this. I'm running VSCode v1.92.0 with Catppuccin v3.15.0 and I can confirm that I can reproduce this issue. 

Interestingly, I'm able to workaround this by targeting the `parameter` and `type` options. See below for the config and a screenshot for how it looks.

`config`
```json
  "catppuccin.customUIColors": {
    "latte": {
      "editorInlayHint.parameterBackground": "red",
      "editorInlayHint.parameterForeground": "crust",
      "editorInlayHint.typeBackground": "pink",
      "editorInlayHint.typeForeground": "crust",
    }
  },
```

`screenshot`
![image](https://github.com/user-attachments/assets/e1e941c8-a050-4fa1-ae6a-a93bdbefffc2)

I'll need to investigate further to find out why the normal `editorInlayHint.foreground` and `editorInlayHint.background` options don't work when the extension isn't enabled.

**Comment by sgoudham** [comment]
Hey again 👋,
I've looked into this again and I think I've figured out why.

The Catppuccin theme defines the following keys:

https://github.com/catppuccin/vscode/blob/39e26a0e1bd5a84899c2b786b9119d6e9174297f/packages/catppuccin-vsc/src/theme/uiColors.ts#L181-L186

`typeForeground`, `typeBackground`, `parameterForeground` & `parameterBackground` are all higher in specificity than just the plain `foreground` and `background`. This means that if you want to override the values for Catppuccin, you'll need to override all of the keys. For example:

```json
"catppuccin.customUIColors": {
  "all": {
    "editorInlayHint.foreground": "#000000",
    "editorInlayHint.background": "#ffffff",
    "editorInlayHint.parameterForeground": "#ffffff",
    "editorInlayHint.parameterBackground": "#000000",
    "editorInlayHint.typeForeground": "#ffffff",
    "editorInlayHint.typeBackground": "#000000",
  }
},
```

<img width="563" alt="image" src="https://github.com/user-attachments/assets/6d3af92f-cf68-400b-a3d6-eda6a6c3e068">

I'll be closing this issue now as it's not necessarily a bug. I think it might be worth looking into ways on "unsetting" the key to avoid duplication but I'll raise another issue for to investigate that, should we want that behaviour.
